winch/codegen/src/regalloc.rs

```diff
@@ -136,7 +136,7 @@ impl RegAlloc {
                 let addr = context.masm.local_address(&slot);
                 context
                     .masm
-                    .store(RegImm::reg(self.scratch), addr, slot.ty.into());
+                    .load(addr, self.scratch, slot.ty.into());
```

Comment:
This path is not heavily exercised at the moment. I'll add more extensive tests once we support `pop` at the Masm layer, which should be once we support all the binary/unary operations since we'd be able to test more complex programs.